_tests/tests_suite_test.go

```diff
@@ -32,52 +32,111 @@ func getRandAppName() string {
+
+	testHome, err = ioutil.TempDir("", "deis-workflow-home")
 	Expect(err).NotTo(HaveOccurred())
+	os.Setenv("HOME", testHome)
 
 	// register the test-admin user
-	register(url, testAdminUser, testAdminPassword, testAdminEmail)
+	sess, err := start("deis register %s --username=%s --password=%s --email=%s", url, testAdminUser, testAdminPassword, testAdminEmail)
+	Expect(err).To(BeNil())
+
+	sess.Wait()
+
+	if strings.Contains(string(sess.Err.Contents()), "must be unique") {
+		// Already registered
+		login(url, testAdminUser, testAdminPassword)
+	} else {
+		Eventually(sess).Should(gexec.Exit(0))
+		Eventually(sess).Should(SatisfyAll(
+			gbytes.Say("Registered %s", testAdminUser),
+			gbytes.Say("Logged in as %s", testAdminUser)))
+	}
+
 	// verify this user is an admin by running a privileged command
-	sess, err := start("deis users:list")
+	sess, err = start("deis users:list")
 	Expect(err).To(BeNil())
 	Eventually(sess).Should(gexec.Exit(0))
 
+	sshDir := path.Join(testHome, ".ssh")
+
 	// register the test user and add a key
 	register(url, testUser, testPassword, testEmail)
-	createKey("deis-test")
-	sess, err = start("deis keys:add ~/.ssh/deis-test.pub")
+	keyPath = createKey(keyName)
+
+	gitSSH = path.Join(sshDir, "git-ssh")
+	ioutil.WriteFile(gitSSH, []byte(fmt.Sprintf("#!/bin/sh\nexec /usr/bin/ssh -v -i %s \"$@\"", keyPath)), 0777)
+
+	sess, err = start("deis keys:add %s.pub", keyPath)
 	Expect(err).To(BeNil())
 	Eventually(sess).Should(gexec.Exit(0))
-	Eventually(sess).Should(gbytes.Say("Uploading deis-test.pub to deis... done"))
+	Eventually(sess).Should(gbytes.Say("Uploading %s.pub to deis... done", keyName))
+
+	time.Sleep(2 * time.Second) // wait for ssh key to propagate
+})
+
+var _ = BeforeEach(func() {
+	var err error
+	var output string
+
+	testRoot, err = ioutil.TempDir("", "deis-workflow-test")
+	Expect(err).NotTo(HaveOccurred())
+
+	os.Chdir(testRoot)
+	output, err = execute(`git clone git@github.com:deis/example-go.git .`)
+	Expect(err).NotTo(HaveOccurred(), output)
+
+})
+
+var _ = AfterEach(func() {
+	err := os.RemoveAll(testRoot)
+	Expect(err).NotTo(HaveOccurred())
 })
 
 var _ = AfterSuite(func() {
 	cancelUserSess, cancelUserErr := cancelSess(url, testUser, testPassword)
 	cancelAdminSess, cancelAdminErr := cancelSess(url, testAdminUser, testAdminPassword)
+
 	Expect(cancelUserErr).To(BeNil())
 	Expect(cancelAdminErr).To(BeNil())
+
 	cancelUserSess.Wait()
 	cancelAdminSess.Wait()
-	Expect(cancelUserSess.ExitCode()).To(BeZero())
-	Expect(cancelAdminSess.ExitCode()).To(BeZero())
-	Expect(cancelUserSess.Out.Contents()).To(ContainSubstring("Account cancelled"))
-	Expect(cancelAdminSess.Out.Contents()).To(ContainSubstring("Account cancelled"))
+
+	//Expect(cancelUserSess.ExitCode()).To(BeZero())
+	//Expect(cancelAdminSess.ExitCode()).To(BeZero())
+	//Expect(cancelUserSess.Out.Contents()).To(ContainSubstring("Account cancelled"))
+	//Expect(cancelAdminSess.Out.Contents()).To(ContainSubstring("Account cancelled"))
```

Comment:
they cause a false negatives sometimes and their success isn't actually required.